File: ObjectiveC/Gauges101/Gauges101/XGExportImageViewController.m
//
//  XGExportImageViewController.m
//  Gauges101
//
//  Copyright © 2016 GrapeCity. All rights reserved.
//

#import "XGExportImageViewController.h"
#import <XuniGaugeKit/XuniGaugeKit.h>

#define LOC(A) NSLocalizedString(A, nil)

@interface XGExportImageViewController ()
@property (weak, nonatomic) IBOutlet XuniRadialGauge *radialGauge;
@property (weak, nonatomic) IBOutlet UIImageView *snapshotDisplay;

@end

@implementation XGExportImageViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    
    _radialGauge.showText = XuniShowTextNone;
    _radialGauge.thickness = 0.6;
    _radialGauge.min = 0;
    _radialGauge.max = 100;
    _radialGauge.value = 25;
    _radialGauge.loadAnimation.duration = 2;
    _radialGauge.updateAnimation.duration = 2.5;
    _radialGauge.isReadOnly = true;
    _radialGauge.showText = XuniShowTextAll;

    [NSTimer scheduledTimerWithTimeInterval:3 target:self selector:@selector(animateNextStep:) userInfo:nil repeats:true];
 
    _snapshotDisplay.hidden = true;
}

-(void)animateNextStep:(id) sender{
    XuniRadialGauge *radialGauge = _radialGauge;
    double random = (arc4random()% 101);
    radialGauge.updateAnimation.easing = [XuniEasing EaseOutElastic];
    radialGauge.value = random;
}


- (IBAction)saveSnapshot:(id)sender {
    if(_snapshotDisplay.hidden)
    {
        UIAlertView *alert = [[UIAlertView alloc] initWithTitle:LOC(@"Warning") message:LOC(@"You should take the snapshot before saving it to Camera Roll") delegate:nil cancelButtonTitle:LOC(@"OK")
                                              otherButtonTitles:nil];
        [alert show];

    }
    else
    {
        UIImageWriteToSavedPhotosAlbum(_snapshotDisplay.image, self, @selector(imageSavedToPhotoAlbum:finishedSavingWithError:contextInfo:), nil);
    }
}

-(void)imageSavedToPhotoAlbum:(UIImage *)image finishedSavingWithError: (NSError *)error contextInfo:(void *)contextInfo{
    NSString *message;
    NSString *title;
    if (!error) {
        title = LOC(@"Success");
        message = LOC(@"Image was saved to Camera Roll succesfully");
    }
    else{
        title = LOC(@"Failure");
        message = [error description];
    }
    UIAlertView *alert = [[UIAlertView alloc] initWithTitle:title message:message delegate:nil cancelButtonTitle:LOC(@"OK") otherButtonTitles:nil];
    [alert show];
}

- (IBAction)takeSnapshot:(id)sender {
    UIImage *image = [[UIImage alloc] init];
    image = [UIImage imageWithData:[_radialGauge getImage]];
    [_snapshotDisplay setImage:image];
    _radialGauge.hidden = true;
    _snapshotDisplay.hidden = false;
    _snapshotDisplay.layer.borderWidth = 10;
    _snapshotDisplay.layer.borderColor = [UIColor blackColor].CGColor;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: Swift/Gauges101/Gauges101/XGExportImageViewController.swift
//
//  XGExportImageViewController.h
//  Gauges101
//
//  Copyright © 2016 GrapeCity. All rights reserved.
//
import UIKit
import XuniCoreDynamicKit

class XGExportImageViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        self.radialGauge.showText = XuniShowText.None
        self.radialGauge.thickness = 0.6
        self.radialGauge.min = 0
        self.radialGauge.max = 100
        self.radialGauge.value = 25
        self.radialGauge.loadAnimation.duration = 2
        self.radialGauge.updateAnimation.duration = 2.5
        self.radialGauge.isReadOnly = true
        self.radialGauge.showText = XuniShowText.All
        NSTimer.scheduledTimerWithTimeInterval(3, target: self, selector: "animateNextStep:", userInfo: nil, repeats: true)
        self.snapshotDisplay.hidden = true
    }

    func animateNextStep(sender: AnyObject) {
        let random: Double = Double(Int(arc4random()) % 101)
        
        //radialGauge.updateAnimation.easing = XuniCoreDynamicKit.XuniEasing.EaseOutElastic()
        radialGauge.value = random
    }

    @IBAction func saveSnapshot(sender: AnyObject) {
        if snapshotDisplay.hidden {
            let alert: UIAlertView = UIAlertView(title: NSLocalizedString("Warning",comment:""), message: NSLocalizedString("You should take the snapshot before saving it to Camera Roll", comment: ""), delegate: nil, cancelButtonTitle: NSLocalizedString("OK",comment:""))
            alert.show()
        }
        else {
            
            UIImageWriteToSavedPhotosAlbum(snapshotDisplay.image!, self, "imageSavedToPhotoAlbum:finishedSavingWithError:contextInfo:",nil)
        }
    }

    func imageSavedToPhotoAlbum(image: UIImage, finishedSavingWithError error: NSError?, contextInfo:UnsafePointer<Void>) {
        var message: String
        var title: String
        if error == nil {
            title = NSLocalizedString("Success",comment:"")
            message = NSLocalizedString("Image was saved to Camera Roll succesfully",comment:"")
        }
        else {
            title = NSLocalizedString("Failure", comment:"")
            message = error!.description
        }
        let alert: UIAlertView = UIAlertView(title: title, message: message, delegate: nil, cancelButtonTitle: NSLocalizedString("OK",comment:""))
        alert.show()
        self.radialGauge.hidden = false
        self.snapshotDisplay.hidden = true
    }

    @IBAction func takeSnapshot(sender: AnyObject) {
        let image: UIImage = UIImage(data: radialGauge.getImage())!
        snapshotDisplay.image = image
        self.radialGauge.hidden = true
        self.snapshotDisplay.hidden = false
        self.snapshotDisplay.layer.borderWidth = 10
        self.snapshotDisplay.layer.borderColor = UIColor.blackColor().CGColor
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBOutlet weak var radialGauge: XuniRadialGauge!
    @IBOutlet weak var snapshotDisplay: UIImageView!
}
//
//  XGExportImageViewController.m
//  Gauges101
//
//  Copyright © 2016 GrapeCity. All rights reserved.
//

import XuniGaugeDynamicKit
//#define LOC(A) NSLocalizedString(A, nil)